WooCommerce/Classes/ViewRelated/ListSelector/ListSelectorViewController.swift

```diff
@@ -56,29 +60,37 @@ UIViewController, UITableViewDataSource, UITableViewDelegate where DataSource.Mo
     }
 
     override func viewWillDisappear(_ animated: Bool) {
-        onDismiss(dataSource.selected)
+        if isMovingFromParent {
+            onDismiss(command.selected)
+        }
         super.viewWillDisappear(animated)
     }
 
+    func reloadData() {
+        tableView.reloadData()
+
+        title = command.navigationBarTitle
+    }
```

Comment:
yes! this is the function to reload data. it's a good idea to use `PublishSubject` for the filter list screen to know when to reload data! lemme try that out now that it's available in `develop` 😃

Reply:
I realized that `ViewController` is needed when it navigates from the filter list selector to the filter option list selector. I haven't thought of another way to pass the navigation controller to `FilterProductListCommand` in https://github.com/woocommerce/woocommerce-ios/pull/2204 - lemme know if you have other ideas!

Reply:
in order not to expose `reloadData()`, were you suggesting adding an Observable to `ListSelectorCommand` that indicates when the UI should reload? if so, I might hold off making this change until the feature is complete, since it involves updating all the implementations of `ListSelectorCommand` and also I was hoping to return a `.just` for implementations whose data never change

Reply:
After looking at #2204 and #2206, let's merge this for now so we don't have a lot of open PRs. 😄 We can make changes in the other PRs if we need to. 

I might have some ideas about #2204. It's working well. I'm thinking `FilterListCommand` may not be the appropriate design pattern for it anymore though. 🤔 I'll continue wrapping my head around it and give you some feedback tomorrow.